tcp_driver/design/tcp_driver-design.md

```diff
@@ -0,0 +1,67 @@
+tcp_driver
+===============
+
+This is the design document for the `tcp_driver` package.
+
+# Purpose / Use cases
+
+<!-- Required -->
+<!-- Things to consider:
+    - Why did we implement this feature? -->
+
+The primary objective of the `tcp_driver` package is to provide wrappers for the
+Transmission Control Protocol (TCP) and the Hyper Text Transport Protocol (HTTP)
+to allow reliable, ordered delivery of a stream of octets (bytes) between applications
+running on hosts (computers) and other devices in a network such as sensors.
+
+The TCP and HTTP protocols are commonly used by sensors to allow configuration and to obtain status reports.
+
+# Design
+
+<!-- Required -->
+<!-- Things to consider:
+    - How does it work? -->
+
+The TCP and HTTP classes allow the synchronous and asynchronous reception of packets.
+
+Both wrappers are implemented using the Boost ASIO library.
```

Comment:
The correct spelling of the library is Boost.Asio (see https://www.boost.org/doc/libs/1_81_0/doc/html/boost_asio.html)

Reply:
Addressed [here](https://github.com/ros-drivers/transport_drivers/pull/84/commits/8871afaee4bab3e6d6928efd9bb0038f5365fe3e)